checkpoint/type.go

```diff
@@ -195,14 +195,14 @@ func Hash(data []byte) []byte {
 func hashBytes(cp *checkpoint.Checkpoint) []byte {
 	ret := make([]byte, 0, len(cp.Governance)+len(cp.Assets)+len(cp.Collateral)+len(cp.NetworkParameters)+len(cp.Delegation)+len(cp.Epoch)+len(cp.Block)+len(cp.Rewards)+len(cp.KeyRotations))
 	// the order in which we append is quite important
+	ret = append(ret, cp.NetworkParameters...)
```

Comment:
is that correct?


```
cpOrder = []types.CheckpointName{
		types.AssetsCheckpoint,         // assets are required for collateral to work, and the vote asset needs to be restored
		types.CollateralCheckpoint,     // without balances, governance (proposals, bonds) are difficult
		types.NetParamsCheckpoint,      // net params should go right after assets and collateral, so vote tokens are restored
		types.GovernanceCheckpoint,     // depends on all of the above
		types.EpochCheckpoint,          // restore epoch information...
		types.DelegationCheckpoint,     // so delegation sequence ID's make sense
		types.PendingRewardsCheckpoint, // pending rewards can basically be reloaded any time
		types.KeyRotationsCheckpoint,   // key rotations can be reloaded any time
	}

```
In checkpoint engine

Reply:
Yup I checked the same function on the core which have a difference order. It’s all good

Reply:
you're right but we shouldn't have it in different order in the same place for.